tests/PHPStan/Analyser/data/array-chunk.php

```diff
@@ -0,0 +1,46 @@
+<?php declare(strict_types = 1);
+
+namespace ArrayChunk;
+
+use function PHPStan\Testing\assertType;
+
+class Foo
+{
+
+	public function generalArrays(array $arr): void
+	{
+		/** @var mixed[] $arr */
+		assertType('array<int, array<int, mixed>>', array_chunk($arr, 2));
+		assertType('array<int, array>', array_chunk($arr, 2, true));
+
+		/** @var array<string, int> $arr */
+		assertType('array<int, array<int, int>>', array_chunk($arr, 2));
```

Comment:
Can't we say for sure that the chunks are always going to be non-empty-array?

Reply:
True. I saw that the result will be empty for an empty array, but the chunks are still non-empty. That actually simplifies the implementation again I think, will adapt it later

Reply:
I adapted the non-empty-array-handling again. the chunk type should be fine to be always non-empty, you're right. the outer type should be non-empty though when the original array was too, I think. amazing that after all the optional complexness I struggle with this xD

thx for checking this closely!